src/webgpu/shader/execution/expression/binary/f32_arithmetic.spec.ts

```diff
@@ -0,0 +1,184 @@
+export const description = `
+Execution Tests for the f32 arithmetic binary expression operations
+`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../../gpu_test.js';
+import { correctlyRoundedThreshold, ulpThreshold } from '../../../../util/compare.js';
+import { kValue } from '../../../../util/constants.js';
+import { f32, TypeF32 } from '../../../../util/conversion.js';
+import { biasedRange, linearRange } from '../../../../util/math.js';
+import { Case, Config, run } from '../expression.js';
+
+import { binary } from './binary.js';
+
+export const g = makeTestGroup(GPUTest);
+
+/* Generates an array of numbers spread over the entire range of 32-bit floats */
+function fullNumericRange(): Array<number> {
+  const numeric_range = Array<number>();
+  numeric_range.concat(biasedRange(kValue.f32.negative.max, kValue.f32.negative.min, 100));
+  numeric_range.concat(
+    linearRange(kValue.f32.subnormal.negative.min, kValue.f32.subnormal.negative.max, 100)
+  );
+  numeric_range.concat(0.0);
+  numeric_range.concat(
+    linearRange(kValue.f32.subnormal.positive.min, kValue.f32.subnormal.positive.max, 100)
+  );
+  numeric_range.concat(biasedRange(kValue.f32.positive.min, kValue.f32.positive.max, 100));
+  return numeric_range;
+}
+
+g.test('addition')
+  .uniqueId('xxxxxxxxx')
+  .specURL('https://www.w3.org/TR/WGSL/#floating-point-evaluation')
+  .desc(
+    `
+Expression: x + y
+Accuracy: Correctly rounded
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const cfg: Config = t.params;
+    cfg.cmpFloats = correctlyRoundedThreshold();
+
+    const truthFunc = (lhs: number, rhs: number): Case => {
```

Comment:
Seems odd that a `truthFunc` would return a `Case`. Don't the other truth functions just return the number?

Reply:
I think the original implementations returned the result, but they have since been changed to return a `Case`, and most of the recent versions I have written return a `Case`. I can see how that drift in their usage could cause some confusion given their name, would changing their names to be something like `makeCase` work for you?